client-storage.js

```diff
@@ -216,21 +216,20 @@ ClientStorage.prototype.get = function (key) {
 ClientStorage.prototype.set = function (key, value, time) {
   if (time) {
     if (this.LSSupport) {
+      let expire = new Date().getTime() + time;
       this.ls.setItem(key, this.__escape(value));
-      setTimeout(() => {
-        this.ls.removeItem(key)
-      }, time)
+      this.ls.setItem(key + "._expiredAt", expire);
     } else if (this.cookies) {
       this.cookies.set(key, this.__escape(value));
-      setTimeout(() => {
-        this.cookies.remove(key, null, window.location.host)
-      }, time)
+      let seconds = time / 1000;
+      document.cookie = key + "=" + value + "; max-age=" + seconds + "; path=/";
```

Comment:
Please use `this.cookies` [library](https://github.com/VeliovGroup/Client-Storage/blob/master/client-storage.js#L78) here. You can update `this.cookies.set` method to support expiration date